scenes/test_string.txt

```diff
@@ -62,6 +62,20 @@
 *gosub test_finish result_expected cslib_ret
 
 
+*comment :::::: INDEX tests ::::::
+*gosub test_start "INDEX #1"
+*temp result_expected 3
+*gosub_scene cslib_string index "ABCDEF" "C"
+*gosub test_finish result_expected cslib_ret
+
+*gosub test_start "INDEX #2"
+*set cslib_catch_bug true
```

Comment:
We could even have a wrapper

*gosub cslib_test catch_bug MODULE FUNC ARGS

that wraps with cslib_catch_bug true and then false.

I think I will add this trick to my preprocessor:
*try SCENE FUNC args

Reply:
Unfortunately I don't think this pattern will work, because you can't re-pass/forward on an arbitrary number of parameters to the target routine.

Reply:
we'd need a bunch of cases - ugly but doable

*if param_count = N
   *gosub_scene param_1 param_2... param_N
   *return

Reply:
That's not really scalable.

Reply:
Lots of routines can take an infinite* number of arguments.

*Within practical reason.

Reply:
We had to do that in production code at Google for function callbacks. I think we had up to 12 arguments. And a function with 12 argument is bad. I understand the sorting / max / min function can take infinite args, but for testing purpose we can limit to say 10. That was just an idea, but we don't have to do it.